extensions/wikia/UserLogin/js/UserLoginAjaxForm.js

```diff
@@ -47,32 +47,63 @@
 
 	/**
 	 * Called when a user has requested a password change
-	 * @param {Object} json Response from server
 	 */
-	UserLoginAjaxForm.prototype.onResetPasswordResponse = function (json) {
-		var callback = this.options.resetpasscallback || '';
-		if (callback && typeof callback === 'function') {
-			// call with current context
-			callback.bind(this, json)();
-		} else {
-			// default implementation
-			$.post(wgScriptPath + '/wikia.php', {
-				controller: 'UserLoginSpecial',
-				method: 'changePassword',
-				format: 'html',
+	UserLoginAjaxForm.prototype.onResetPasswordResponse = function () {
+		$.nirvana.sendRequest({
+			controller: 'UserLoginSpecial',
+			method: 'changePassword',
+			format: 'html',
+			data: {
 				username: this.inputs.username.val(),
 				password: this.inputs.password.val(),
+				loginToken: this.inputs.loginToken.val(),
 				returnto: this.inputs.returnto.val(),
 				fakeGet: 1
-			}, this.retrieveTemplateCallback.bind(this));
+			},
+			callback: this.setupChangePasswordForm.bind(this)
+		});
+	};
+
+	/**
+	 * Replace form's content based on HTML sent from the server.
+	 * @param {string} html
+	 */
+	UserLoginAjaxForm.prototype.setupChangePasswordForm = function (html) {
+		var content, form, heading, modal, contentBlock, duration;
+
+		content = $('<div>').hide().append(html);
```

Comment:
This should be safe since we're not getting the HTML from the DOM, where the user could have manipulated it prior to retrieval. We're getting the HTML from a server request, and the HTML is generated from templates where all the UCG has been escaped. So we should be fine here.